modules/ximgproc/src/structured_edge_detection.cpp

```diff
@@ -810,6 +815,93 @@ class StructuredEdgeDetectionImpl : public StructuredEdgeDetection
         imsmooth( dstM.reshape(1, dst.rows), 1 ).copyTo(dst);
     }
 
+protected:
+    struct RandomForest;
+
+    /**
+     * Load model from a file.
+     * \param filename [in] Model file name.
+     * \param rf       [out] Model where parameters are saved.
+     */
+    void loadModelFromBinaryFile(const String &filename, struct RandomForest &rf)
+    {
+        FILE *f = fopen(filename.c_str(), "rb");
+        CV_Assert(f);
+
+        int help;
+        size_t ret;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.stride = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.shrinkNumber = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.patchSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.patchInnerSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfGradientOrientations = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.gradientSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.regFeatureSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.ssFeatureSmoothingRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.gradientNormalizationRadius = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.selfsimilarityGridSize = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfTrees = help;
+
+        ret = fread(&help, sizeof(int), 1, f);
+        rf.options.numberOfTreesToEvaluate = help;
+
+        rf.options.numberOfOutputChannels = 2*(rf.options.numberOfGradientOrientations+1)+3;
+
+        size_t len;
+        uint64_t len_;
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.childs.resize(len);
+        ret = fread(rf.childs.data(), sizeof(int), len, f);
+
+        ret = fread(&len_, sizeof(uint64_t), 1, f);
+        len = (size_t)len_;
+        rf.featureIds.resize(len);
+        ret = fread(rf.featureIds.data(), sizeof(int), len, f);
```

Comment:
We definitely should avoid such code for custom formats.

Consider to use BASE64 encoding or HDF5 binary formats.

Reply:
@alalek 
I find that you tried to add HDF5 support to OpenCV (https://github.com/opencv/opencv/compare/master...alalek:hdf5)

But it seems that the work is stopped. Maybe I could reuse your work.

Reply:
There is hdf module in opencv_contrib: https://github.com/opencv/opencv_contrib/tree/3.3.0/modules/hdf